src/main/java/com/jakewharton/retrofit2/adapter/rxjava2/RxJava2CallAdapter.java

```diff
@@ -69,13 +69,13 @@
       return observable.toFlowable(BackpressureStrategy.LATEST);
     }
     if (isSingle) {
-      return observable.toSingle();
+      return observable.single(null);
```

Comment:
What would work for now would be `return observable.firstElement().toSingle();` Although unit tests are failing since `firstElement` had a bug that it did not deliver exceptions to the plugin which is fixed with https://github.com/ReactiveX/RxJava/pull/4590

RC4 would also bring a method `observable.singleOrError()` which brings back the old `toSingle()` behavior.